Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -63,6 +81,11 @@ public object Deserialize(IIonReader reader)
 
         public void Serialize(IIonWriter writer, object item)
         {
+            if (writer == null)
+            {
+                throw new ArgumentException("Cannot serialize with a null Ion writer");
```

Comment:
Should be `NullReferenceException1` if you really want to test this kind of thing, which I don't personally value.

Reply:
Changed to `NullReferenceException`. This check is here because the method is public.